tools/src/main/java/org/apache/kafka/trogdor/workload/RoundTripWorker.java

```diff
@@ -286,7 +290,7 @@ public void run() {
                         pollInvoked++;
                         ConsumerRecords<byte[], byte[]> records = consumer.poll(50);
                         for (ConsumerRecord<byte[], byte[]> record : records.records(TOPIC_NAME)) {
-                            int messageIndex = ByteBuffer.wrap(record.key()).getInt();
+                            int messageIndex = ByteBuffer.wrap(record.key()).order(ByteOrder.LITTLE_ENDIAN).getInt();
```

Comment:
Was this the fix for the failing tests? The JIRA mentions this: "Currently RoundTripWorkload is unable to get the sequence number of the keys that it produced." I'm having some trouble understanding how that description is mapping to the patch.

Reply:
The previous code was using 8-byte sequence numbers rather than 4, so the fix would have been a little different to try to work with that generator.  However, the previous code wasn't able to properly change position back to an older position and get the same value as last time, which was also a problem in some contexts (although maybe not this one)